src/pytorch_lightning/utilities/enums.py

```diff
@@ -15,52 +15,32 @@
 from __future__ import annotations
 
 import os
-from enum import Enum
+from typing import TYPE_CHECKING
 
-from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from lightning_utilities.core.enums import StrEnum
 
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
 
-class LightningEnum(str, Enum):
-    """Type of any enumerator with allowed comparison to string invariant to cases."""
+if TYPE_CHECKING:
+    from enum import Enum
 
-    @classmethod
-    def from_str(cls, value: str) -> LightningEnum | None:
-        statuses = cls.__members__.keys()
-        for st in statuses:
-            if st.lower() == value.lower():
-                return cls[st]
-        return None
+    # re-defined because `mypy` infers `StrEnum` as `Any`
```

Comment:
I don't understand why mypy does not infer the `StrEnum` type if it's from a library. This workaround seems fine for the moment.